From cfe78f5cb9bcba10f9f71e84aaef407c92bca36a Mon Sep 17 00:00:00 2001
From: Colin Guthrie <colin@mageia.org>
Date: Fri, 4 Nov 2011 00:28:35 +0000
Subject: [PATCH 502/509] mageia: Change the unit for prefdm.service to make
 it work a bit more like Fedora

Do not wait for some of the Fedora units that we have traditionally not
waited for in Mageia or Mandriva before it.
---
 units/mageia/prefdm.service |   15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/units/mageia/prefdm.service b/units/mageia/prefdm.service
index c85a7a9..db41294 100644
--- a/units/mageia/prefdm.service
+++ b/units/mageia/prefdm.service
@@ -7,15 +7,16 @@
 
 [Unit]
 Description=Display Manager
-After=livesys-late.service rc-local.service systemd-user-sessions.service
-After=network.target acpid.service fs.service haldaemon.service
+After=livesys-late.service systemd-user-sessions.service
 
-# Do not stop plymouth, it is done in prefdm if required
-Conflicts=plymouth-quit.service
-After=plymouth-quit.service
+# On Mageia gdm/X11 is on tty1. We explicitly cancel the getty here to
+# avoid any races around that.
+# Do not stop plymouth, it is done in prefdm if required (or left to the dm)
+Conflicts=getty@tty1.service plymouth-quit.service
+After=getty@tty1.service plymouth-quit.service
 
 [Service]
-ExecStart=/etc/X11/prefdm
-Type=forking
+ExecStart=/etc/X11/prefdm -nodaemon
 Restart=always
 RestartSec=0
+IgnoreSIGPIPE=no
-- 
1.7.10.3

